### PR TITLE
Retrieve content via absolute URL lookup

### PR DESCRIPTION
**Pull request by mparent61**
@MichaelButkovic  - This works, but I'm a REST novice, so please advise. 

- Using the list method to return a single object (which seems weird) 
- Passing the URL by query param seems cleaner than including it as part of the url path, since it will contain backslashes. 

A separate endpoint seemed like a cleaner approach, after exploring adding a filter to existing Content endpoint.

### CODE REVIEW

**Comment by mparent61** [comment]
@clowe why not...

File: bulbs/api/views.py [hunk 1]
```diff
@@ -439,6 +441,21 @@ def list(self, request):
         return Response(dict(results=results))
 
 
+class ContentResolveViewSet(viewsets.ReadOnlyModelViewSet):
+    """Retrieve content corresponding to absolute URL"""
+    model = Content
+
+    def list(self, request):
+        if "url" in get_query_params(self.request):
```
Comment (MichaelButkovic) [inline bulbs/api/views.py]: May want to throw a 400 or something if `url` isn't found.

**Comment by MichaelButkovic** [comment]
LGTM

File: bulbs/api/views.py (hunk 1 above)
Comment (mparent61) [inline bulbs/api/views.py]: good idea

**Comment by mparent61** [comment]
@spra85 look OK?

File: tests/api/test_content_api.py [hunk 2]
```diff
@@ -581,3 +581,27 @@ def test_search(self):
...
+
+
+class TestContentResolveAPI(BaseAPITestCase):
+
+    def test_resolve_url(self):
+        make_content(id=123)
+        r = self.api_client.get(reverse("content-resolve-list"),
+                                dict(url="/r/123"), format="json")
+        self.assertEqual(r.status_code, 200)
+        self.assertEqual(123, r.data['id'])
```
Comment (spra85) [inline tests/api/test_content_api.py]: Should we add one more test that's not just the `/r/<id>` shorthand, e.g. http://www.theonion.com/article/man-arriving-late-forced-use-excuse-he-was-saving--51824

Comment (spra85) [inline tests/api/test_content_api.py]: Also, maybe one just to be safe with a bunch of UTM or other parameters tacked onto the end.

**Comment by spra85** [comment]
Looks good, just the couple of thoughts on a few add'l tests to make sure we're buttoned up

File: tests/api/test_content_api.py (hunk 2 above)
Comment (mparent61) [inline tests/api/test_content_api.py]: Yeah i'll throw the params on.
The longer URLs are property-specific and not defined in django-bulbs, but might make sense to mock one up.

**Pull request merged**